### DISCUSSION TITLE
Using external image when using draw

### DISCUSSION THREAD

**Discussion by vicocamacho** [ImageMagick/ImageMagick#2175] [Help] [answered]
Hi!

I'm using ImageMagick 7.0.10-19 Q16 x86_64 2020-06-17 https://imagemagick.org on Ubuntu 18.04 and I was wondering if it is possible to use images from external resources when using `-draw` something like this:

```
magick mogrify -draw "image SrcOver 0,0,0,0 'https://somewebsite.com/exterior_siding_vinyl_Georgia_Pacific_Grey_20200515_0001.png'" /tmp/mini_magick20200619-13025-4kegl4.jpg~
```

because I'm getting an error for non-conforming drawing primitive.

Is there a way to do this? Any pointers would be appreciated.

Thank you!

**Comment by fmw42** [comment]
I do not think it will work. I tried. First I edited my policy.xml file to give read|write permission to HTTPS and also to @. Then I tried both:

`magick mogrify -format jpg -draw "image srcover 0,0,0,0 'https://imagemagick.org/image/label.gif'" /Users/fred/desktop/testing/lena.png`

and

```
magick 'https://imagemagick.org/image/label.gif' GIF:- | \
magick mogrify -format jpg -draw "image srcover 0,0,0,0 '@-'" /Users/fred/desktop/testing/lena.png
```

But they did not work. I got your same message.  So the only way I can think of to do what you want is to download the file and then use it in magick mogrify:

```
magick 'https://imagemagick.org/image/label.gif' x.gif
magick mogrify -format jpg -draw "image srcover 0,0,0,0 'x.gif'" /Users/fred/desktop/testing/lena.png
```

I suspect that mogrify is too old for use of HTTPS files that way. But an ImageMagick developer would have to comment.

**Reply by vicocamacho** [reply]
Thanks for the reply, would it be the same if the endpoints where in `http`?

**Comment by fmw42** [comment]
I tried that and still get the same message. So I think you will have to do it the long way as I described above.

But one of the IM developers will have to comment about this to be sure I have not done something wrong.

**Comment by snibgo** [comment]
It works fine for me, IM v7.0.8-64 on Windows 8.1.
```
magick mogrify -draw "image SrcOver 0,0,0,0 'https://imagemagick.org/image/label.gif'" x.png
```

**Reply by fmw42** [reply]
Interesting!?

**Comment by urban-warrior** [comment, marked as answer]
The image drawing element no longer supports a http(s) URI.  It is considered a security risk.  Instead, for example, try

```
wget https://imagemagick.org/image/label.gif
magick mogrify -draw "image SrcOver 0,0,0,0 'label.gif' x.png
```
We'll add a patch to improve the exception that is thrown.
